ngraph/core/include/ngraph/partial_shape.hpp

```diff
@@ -233,6 +233,7 @@ namespace ngraph
         // True if the shape is static.
         mutable enum class ShapeType {
             SHAPE_IS_UNKNOWN,
+            SHAPE_IS_UPDATED,
```

Comment:
Please add description for each ShapeType (purpose of each type, how we need to maintain them)

Reply:
done